spark-connector/spark3.3/build.gradle.kts

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+plugins {
+  `maven-publish`
+  id("java")
+  id("idea")
+  alias(libs.plugins.shadow)
+}
+
+repositories {
+  mavenCentral()
+}
+
+val scalaVersion: String = project.properties["scalaVersion"] as? String ?: extra["defaultScalaVersion"].toString()
+val sparkVersion: String = "3.3.1"
+val sparkMajorVersion: String = sparkVersion.substringBeforeLast(".")
+val icebergVersion: String = "1.4.3"
```

Comment:
Even though this pr isn't ready yet, I'd like to ask, why use `1.4.3` instead of being consistent with gravitino server's iceberg version?
Because I'm cherry-picking patches of `RESTCatalog` from the community branch to the internal branch, I worry about the impact of inconsistent iceberg versions.

Reply:
gravitino server use `1.3.1` which is not supported in `Spark3.5`. and the Iceberg version here not infect the Gravitino server Iceberg version. and spark connector should support multi Iceberg versions.

Reply:
> gravitino server use `1.3.1` which is not supported in `Spark3.5`. and the Iceberg version here not infect the Gravitino server Iceberg version. and spark connector should support multi Iceberg versions.

ok.